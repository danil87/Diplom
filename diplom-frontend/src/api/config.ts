import {
  BaseQueryFn,
  FetchArgs,
  fetchBaseQuery,
  FetchBaseQueryError,
} from '@reduxjs/toolkit/query'
import { AccessToken } from 'type'

export const getBaseQuery = (baseUrl: string) => {
  const baseQuery = fetchBaseQuery({
    baseUrl,
    credentials: 'include',
    mode: 'cors',
    prepareHeaders: headers => {
      const accessToken = localStorage.getItem('access')

      if (accessToken) {
        headers.set('authorization', `Bearer ${accessToken}`)
      }

      return headers
    },
  })

  const baseQueryFn: BaseQueryFn<
    string | FetchArgs,
    unknown,
    FetchBaseQueryError
  > = async (args, api, extraOptions) => {
    const result = await baseQuery(args, api, extraOptions)

    if (result.error && result.error.status === 401) {
      const accessTokenData = await baseQuery(
        {
          url: 'http://localhost:8000/api/v1/token/refresh/',
          method: 'POST',
        },
        api,
        extraOptions
      )

      if (accessTokenData.error) {
        localStorage.removeItem('access')
        window.location.reload()
        return result
      }

      const { access } = accessTokenData.data as AccessToken
      localStorage.setItem('access', access)

      return await baseQuery(args, api, extraOptions)
    }

    return result
  }

  return baseQueryFn
}

export const baseQueryFnForAuth: BaseQueryFn<
  string | FetchArgs,
  unknown,
  FetchBaseQueryError
> = async (args, api, extraOptions) => {
  const baseQuery = getBaseQuery('http://localhost:8000/api/v1/auth')

  const result = await baseQuery(args, api, extraOptions)

  if (
    !result.error &&
    result.data &&
    typeof result.data === 'object' &&
    'access' in result.data
  ) {
    const { access } = result.data as { access: string }
    localStorage.setItem('access', access)
  }

  return result
}
