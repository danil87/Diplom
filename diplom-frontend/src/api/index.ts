export { authApi } from './auth-api'
export { equipmentApi } from './equipment-api'
export { equipmentAssignmentApi } from './equipment-assignment'
export { manufacturerApi } from './manufacturer-api'
export { reportApi } from './report-api'
export { userApi } from './user-api'
