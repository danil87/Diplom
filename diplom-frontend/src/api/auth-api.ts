import { createApi } from '@reduxjs/toolkit/query/react'
import { LoginData } from 'components/login-from'
import { RegisterData } from 'components/register-from'
import { AccessToken } from 'type'

import { baseQueryFnForAuth } from './config'

export const authApi = createApi({
  reducerPath: 'authApi',
  baseQuery: baseQueryFnForAuth,
  endpoints: builder => ({
    signIn: builder.mutation<AccessToken, LoginData>({
      query: body => ({
        url: 'login',
        method: 'POST',
        body,
      }),
    }),
    register: builder.mutation<void, RegisterData>({
      query: body => ({
        url: 'create',
        method: 'POST',
        body,
      }),
    }),
    signOut: builder.mutation<void, void>({
      query: () => ({
        url: 'logout',
        method: 'POST',
      }),
    }),
  }),
})
