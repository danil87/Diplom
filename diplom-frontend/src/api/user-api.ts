import { createApi } from '@reduxjs/toolkit/query/react'
import { QueryResponse, User } from 'type'

import { getBaseQuery } from './config'

export const userApi = createApi({
  reducerPath: 'userApi',
  baseQuery: getBaseQuery('http://localhost:8000/api/v1/users'),
  endpoints: builder => ({
    getUser: builder.query<User, void>({
      query: () => 'user',
    }),
    getUsers: builder.query<QueryResponse<User>, string>({
      query: params => `?${params}`,
    }),
    deleteUser: builder.mutation<void, number>({
      query: id => ({
        url: `${id}`,
        method: 'DELETE',
      }),
    }),
  }),
})
