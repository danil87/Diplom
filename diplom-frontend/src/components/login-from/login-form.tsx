import { yupResolver } from '@hookform/resolvers/yup'
import LockOutlinedIcon from '@mui/icons-material/LockOutlined'
import LoadingButton from '@mui/lab/LoadingButton'
import { Avatar, Box, Grid, TextField, Typography } from '@mui/material'
import { authApi } from 'api'
import { PasswordInput } from 'components/password-input'
import { FC, useEffect, useState } from 'react'
import { useForm } from 'react-hook-form'
import { useTranslation } from 'react-i18next'
import { Link, useLocation, useNavigate } from 'react-router-dom'
import { loginScheme } from 'utils/scheme'

import { CustomAlert, Wrapper } from './login-form.styled'
import { LoginData } from './type'

export const LoginForm: FC = () => {
  const { state } = useLocation()
  const [errMes, setErrMes] = useState<string>('')

  const {
    register,
    handleSubmit,
    formState: { errors, isValid },
  } = useForm<LoginData>({
    mode: 'all',
    resolver: yupResolver(loginScheme),
  })
  const navigate = useNavigate()
  const { t } = useTranslation()
  const [signIn, { isLoading, isSuccess, isError, error }] =
    authApi.useSignInMutation()

  useEffect(() => {
    if (isSuccess) {
      navigate('/')
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isSuccess])

  useEffect(() => {
    if (error && 'data' in error) {
      setErrMes(Object.values(error.data as object)[0] as string)
    }
  }, [error])

  return (
    <>
      {(isError || state) && (
        <CustomAlert severity={isError ? 'error' : 'success'}>
          {error ? errMes : state && t('auth.registerSuccess')}
        </CustomAlert>
      )}
      <Wrapper>
        <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component='h1' variant='h5'>
          {t('auth.signIn')}
        </Typography>
        <Box
          component='form'
          onSubmit={handleSubmit(signIn)}
          noValidate
          sx={{ mt: 1 }}
        >
          <TextField
            {...register('username')}
            error={!!errors.username?.message}
            helperText={errors.username?.message && t(errors.username?.message)}
            margin='normal'
            fullWidth
            label={t('auth.username')}
          />
          <PasswordInput
            {...register('password')}
            error={!!errors.password?.message}
            helperText={errors.password?.message && t(errors.password?.message)}
          />

          <LoadingButton
            loading={isLoading}
            disabled={!isValid}
            type='submit'
            fullWidth
            variant='contained'
            sx={{ mt: 3, mb: 2 }}
          >
            {t('auth.signIn')}
          </LoadingButton>
          <Grid container>
            <Grid item>
              <Link to='/sign-up'>{t('auth.dontHaveAccount')}</Link>
            </Grid>
          </Grid>
        </Box>
      </Wrapper>
    </>
  )
}
