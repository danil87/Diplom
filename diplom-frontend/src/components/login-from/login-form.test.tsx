import { fireEvent, screen } from '@testing-library/react'
import { userApi } from 'api/user-api'
import { act } from 'react'
import { renderWithRouterAndProvider } from 'tests/helpers'

import { LoginForm } from './login-form'

describe('Login form tests', () => {
  let mockGetUser: ReturnType<typeof jest.spyOn>

  beforeEach(() => {
    jest.resetAllMocks()
    mockGetUser = jest.spyOn(userApi, 'useGetUserQuery')
  })

  it('Button is disabled', () => {
    mockGetUser.mockReturnValue({ isLoading: false, data: null })

    renderWithRouterAndProvider(<LoginForm />)

    const button = screen.getByTestId(/login-button/)

    expect(screen.getByTestId(/username-input/)).toBeInTheDocument()
    expect(screen.getByTestId(/password-input/)).toBeInTheDocument()
    expect(button).toBeDisabled()
    expect(screen.getByTestId(/form-login/)).toMatchSnapshot()
  })

  it('The password has less than 8 characters', async () => {
    mockGetUser.mockReturnValue({ isLoading: false, data: null })

    renderWithRouterAndProvider(<LoginForm />)

    const usernameInput: HTMLInputElement = screen.getByTestId(/username-input/)
    const passwordInput: HTMLInputElement = screen.getByTestId(/password-input/)
    const button = screen.getByTestId(/login-button/)

    await act(async () => {
      fireEvent.input(usernameInput, { target: { value: 'admin' } })
      fireEvent.input(passwordInput, { target: { value: '1111' } })
      fireEvent.blur(usernameInput)
      fireEvent.blur(passwordInput)
    })

    expect(usernameInput.value).toBe('admin')
    expect(passwordInput.value).toBe('1111')
    expect(button).toBeDisabled()
    expect(screen.getByTestId(/form-login/)).toMatchSnapshot()
  })

  it('The password has 8 characters', async () => {
    mockGetUser.mockReturnValue({ isLoading: false, data: null })

    renderWithRouterAndProvider(<LoginForm />)

    const usernameInput: HTMLInputElement = screen.getByTestId(/username-input/)
    const passwordInput: HTMLInputElement = screen.getByTestId(/password-input/)
    const button = screen.getByTestId(/login-button/)

    await act(async () => {
      fireEvent.change(usernameInput, { target: { value: 'admin' } })
      fireEvent.change(passwordInput, { target: { value: '11111111' } })
      fireEvent.blur(usernameInput)
      fireEvent.blur(passwordInput)
    })

    expect(usernameInput.value).toBe('admin')
    expect(passwordInput.value).toBe('11111111')
    expect(button).not.toBeDisabled()
    expect(screen.getByTestId(/form-login/)).toMatchSnapshot()
  })

  it('The password has 8 characters and empty username', async () => {
    mockGetUser.mockReturnValue({ isLoading: false, data: null })

    renderWithRouterAndProvider(<LoginForm />)

    const passwordInput: HTMLInputElement = screen.getByTestId(/password-input/)
    const button = screen.getByTestId(/login-button/)

    await act(async () => {
      fireEvent.change(passwordInput, { target: { value: '11111111' } })
      fireEvent.blur(passwordInput)
    })

    expect(screen.getByTestId(/username-input/)).toBeInTheDocument()
    expect(passwordInput.value).toBe('11111111')
    expect(button).toBeDisabled()
    expect(screen.getByTestId(/form-login/)).toMatchSnapshot()
  })
})
