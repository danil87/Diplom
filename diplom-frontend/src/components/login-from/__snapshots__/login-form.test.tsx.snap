// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form tests Button is disabled 1`] = `
<form
  class="MuiBox-root css-164r41r"
  data-testid="form-login"
  novalidate=""
>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for=":r0:"
      id=":r0:-label"
    >
      auth.username
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="username-input"
        id=":r0:"
        name="username"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-yjsfm1"
        >
          <span>
            auth.username
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-1iledgx-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for="outlined-adornment-password"
    >
      auth.password
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="password-input"
        name="password"
        type="password"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="toggle password visibility"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-yjsfm1"
        >
          <span>
            auth.password
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth Mui-disabled MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1qudo6r-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
    data-testid="login-button"
    disabled=""
    id=":r1:"
    tabindex="-1"
    type="submit"
  >
    auth.signIn
  </button>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <a
        href="/sign-up"
      >
        auth.dontHaveAccount
      </a>
    </div>
  </div>
</form>
`;

exports[`Login form tests The password has 8 characters 1`] = `
<form
  class="MuiBox-root css-164r41r"
  data-testid="form-login"
  novalidate=""
>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r4:"
      id=":r4:-label"
    >
      auth.username
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="username-input"
        id=":r4:"
        name="username"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            auth.username
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-1iledgx-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="outlined-adornment-password"
    >
      auth.password
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="password-input"
        name="password"
        type="password"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="toggle password visibility"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            auth.password
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1qudo6r-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
    data-testid="login-button"
    id=":r5:"
    tabindex="0"
    type="submit"
  >
    auth.signIn
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <a
        href="/sign-up"
      >
        auth.dontHaveAccount
      </a>
    </div>
  </div>
</form>
`;

exports[`Login form tests The password has 8 characters and empty username 1`] = `
<form
  class="MuiBox-root css-164r41r"
  data-testid="form-login"
  novalidate=""
>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for=":r6:"
      id=":r6:-label"
    >
      auth.username
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="username-input"
        id=":r6:"
        name="username"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-yjsfm1"
        >
          <span>
            auth.username
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-1iledgx-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="outlined-adornment-password"
    >
      auth.password
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="password-input"
        name="password"
        type="password"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="toggle password visibility"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            auth.password
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth Mui-disabled MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1qudo6r-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
    data-testid="login-button"
    disabled=""
    id=":r7:"
    tabindex="-1"
    type="submit"
  >
    auth.signIn
  </button>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <a
        href="/sign-up"
      >
        auth.dontHaveAccount
      </a>
    </div>
  </div>
</form>
`;

exports[`Login form tests The password has less than 8 characters 1`] = `
<form
  class="MuiBox-root css-164r41r"
  data-testid="form-login"
  novalidate=""
>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r2:"
      id=":r2:-label"
    >
      auth.username
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="username-input"
        id=":r2:"
        name="username"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            auth.username
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth css-1iledgx-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="outlined-adornment-password"
    >
      auth.password
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="true"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="password-input"
        name="password"
        type="password"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="toggle password visibility"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            auth.password
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
    >
      validate.passwordLength
    </p>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth Mui-disabled MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1qudo6r-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
    data-testid="login-button"
    disabled=""
    id=":r3:"
    tabindex="-1"
    type="submit"
  >
    auth.signIn
  </button>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <a
        href="/sign-up"
      >
        auth.dontHaveAccount
      </a>
    </div>
  </div>
</form>
`;
